sonarts-core/docs/rules/no-collection-index-overwrite.md

```diff
@@ -0,0 +1,49 @@
+# no-collection-index-overwrite
+
+It is highly suspicious when a value is saved for a key or index and then unconditionally 
+overwritten. Such replacements are likely in error.
+
+## Noncompliant Code Example
+
+```typescript
+function arr() {
```

Comment:
@saberduck create subtask for TypeScript in RSPEC with examples
Btw I think it's better to avoid functions, or at least return variable to not have deadstore

Reply:
@saberduck you don't need to put every collection to examples, 1-2 is enough